server/src/main/java/org/candlepin/controller/CandlepinPoolManager.java

```diff
@@ -1917,11 +1917,84 @@ public Subscription fabricateSubscriptionFromPool(Pool pool) {
             throw new IllegalArgumentException("pool is null");
         }
 
+        Long poolQuantity = pool.getQuantity();
+        Long multiplier = 1L;
+
+        if (pool.getProduct() != null) {
+            multiplier = pool.getProduct().getMultiplier();
+        }
+        else {
+            log.error("Master product for the pool %s is null", pool.getId());
+        }
+
+
+        /**
+         * The following code reconstructs Subscription quantity from the Pool quantity.
+         * To understand it, it is important to understand how pool (the parameter)
+         * is created in candlepin from a source subscription.
+         * The pool has quantity was computed from
+         * source subscription quantity and was multiplied by product.multiplier.
+         * To reconstruct subscription, we must therefore divide the quantity of the pool
+         * by the product.multiplier.
+         * It's not easy to find COMPLETE code related to the conversion of
+         * subscription to the pool. There is a method convertToMasterPool in this class,
+         * that should do part of that (multiplication is not there).
+         * But looking at its javadoc, it directly instructs callers of the
+         * convertToMasterPool method to override quantity with method
+         * PoolRules.calculateQuantity (when browsing the code that calls convertToMasterPool,
+         * the calculateQuantity is usually called after convertToMasterPool).
+         * The method PoolRules.calculateQuantity does the actual
+         * multiplication of pool.quantity by pool.product.multiplier.
+         * It seems that we also need to account account for
+         * instance_multiplier (again logic is in calculateQuantity). If the attribute
+         * is present, we must further divide the poolQuantity by
+         * product.getAttributeValue("instance_multiplier").
+         */
+        if (poolQuantity != null && multiplier != null && pool.getProduct() != null) {
+            if (poolQuantity % multiplier != 0) {
```

Comment:
I added the check for both multiplier and instMult (instance_multiplier) and unit test for instance_multiplier=0.

I haven't added unit test for multiplier=0 because it's actually not possible to get zero/null multiplier. 
```java
    public void setMultiplier(Long multiplier) {
        if (multiplier == null) {
            this.multiplier = 1L;
        }
        else {
            this.multiplier = Math.max(1L, multiplier);
        }
    }
```
I also fixed formatting anchors.